src/components/inputs/localized-money-input/localized-money-input.js

```diff
@@ -0,0 +1,319 @@
+    onFocus: PropTypes.func,
+    isDisabled: PropTypes.bool,
+    currenciesControl: PropTypes.node,
+    hasError: PropTypes.bool,
+    hasWarning: PropTypes.bool,
+    placeholder: PropTypes.string,
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+    error: PropTypes.node,
+    warning: PropTypes.node,
+  };
+
+  handleChange = event => {
+    // We manipulate the event to add the currency to the target.
+    // That way the users  can read
+    // event.target.currency and event.target.value to determine the next value.
+    //
+    // We only need this information for the story, the MC application code will
+    // never need to access the information in such an inconvenient way, as
+    // Formik can deal with a name like "foo.en" and sets the value correctly.
+    // We can't use this as we aren't guaranteed a name in the story as the user
+    // might clear it using the knob, and then we can't parse the currency from
+    // the input name anymore.
+    //
+    // eslint-disable-next-line no-param-reassign
+    event.target.currency = this.props.currency;
+    this.props.onChange(event);
+  };
+
+  render() {
+    return (
+      <Spacings.Stack scale="xs">
+        <div className={styles.fieldContainer}>
+          <MoneyInput
+            id={this.props.id}
+            name={this.props.name}
+            value={this.props.value}
+            onChange={this.handleChange}
+            onBlur={this.props.onBlur}
+            isDisabled={this.props.isDisabled}
+            placeholder={this.props.placeholder}
+            hasError={this.props.hasError}
+            hasWarning={this.props.hasWarning}
+            {...filterDataAttributes(this.props)}
+          />
+        </div>
+        <div className={styles.commandsContainer}>
+          <div className={styles.commandsLeft}>
+            {do {
+              if (this.props.error) <div>{this.props.error}</div>;
+              else if (this.props.warning) <div>{this.props.warning}</div>;
+              else this.props.currenciesControl;
+            }}
+          </div>
+        </div>
+        {(this.props.error || this.props.warning) &&
+          this.props.currenciesControl}
+      </Spacings.Stack>
+    );
+  }
+}
+
+export class LocalizedMoneyInput extends React.Component {
+  static displayName = 'LocalizedMoneyInput';
+
+  static propTypes = {
+    id: PropTypes.string,
+    name: PropTypes.string,
+    // then input doesn't accept a "currencies" prop, instead all possible
+    // currencies have to exist (with empty or filled strings) on the value:
+    //   { EUR: '12.00', USD: '14.23', CAD: '13.70' }
+    value: PropTypes.objectOf(PropTypes.string).isRequired,
+    onChange: PropTypes.func,
+    selectedCurrency: PropTypes.string.isRequired,
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    hideExpansionControls: PropTypes.bool,
+    isDefaultExpanded: (props, propName, componentName, ...rest) => {
```

Comment:
@dferber90 is this consistent with https://github.com/commercetools/ui-kit/pull/389?

Reply:
Oh nice catch! Yeah, this needs to be changed indeed! I totally missed it :/

Reply:
I believe it should be 

```js
    hideCurrencyExpansionControls: PropTypes.bool,
    isDefaultExpanded
```

right?

Reply:
> so `isDefaultExpanded` will be `defaultExpandCurrencies`?

Exactly!

And `hideExpansionControls` should be renamed to `hideCurrencyControls`.

> this doesn't sound like a boolean prop, or did I misunderstand the change here

That's true, it doesn't sound like a boolean prop.

There is an "unspoken" convention in React where props which start with `default` will be respected only as an initial value. Changes to that prop will be ignored. The `default` prefix is used when that prop results in an [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html) behaviour for that aspect. An example of this is `input` which can be used as a controlled component `<input value="foo" onChange={fn} />` or as an uncontrolled component `<input defaultValue="foo" />`.

I found the `default` prefix to be of higher importance than having the boolean prefix, as it's more important to signal that this is a prop which is only used to determine the initial state than it is to signal that it contains a boolean.

I could have named it `defaultAreCurrenciesExpanded` or something, but I felt that it's good enough and more readable as the names are right now.

Reply:
Thanks for the clarification

Reply:
What we want is:

```js
defaultExpandCurrencies // replaces isDefaultExpanded
hideCurrencyExpansionControls // replaces hideExpansionControls
```

Reply:
done

Reply:
There are still references to `isDefaultExpanded` in the README and in a spec file 🔍

Reply:
sorry forgot to push :D